lib/spack/spack/test/spec_dag.py

```diff
@@ -189,7 +189,7 @@ def test_conditional_dep_with_user_constraints():
         assert ('y@3' in spec)
 
 
-@pytest.mark.usefixtures('mutable_mock_repo')
+@pytest.mark.usefixtures('mutable_mock_repo', 'config')
```

Comment:
@tldahlgren I added this here because otherwise tests are failing for me locally. Not sure why they pass on Travis. This is a cherry-pick of #15655

Reply:
Okay.  I'll re-run the tests locally with your new version.